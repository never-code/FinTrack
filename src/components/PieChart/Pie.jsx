import React, { useEffect } from 'react';
import Highcharts from 'highcharts';
import accessibility from 'highcharts/modules/variable-pie';

accessibility(Highcharts);
const Pie = ({ interest = 0, principal = 0, backgroundColor }) => {
  console.log(interest, principal, backgroundColor, "##");
  useEffect(() => {
    Highcharts.chart('container', {
      chart: {
        plotBackgroundColor: null,
        plotBorderWidth: null,
        plotShadow: false,
        type: 'pie',
        backgroundColor,
      },
      colors: ['#FBD94F', '#0170F3'],
      legend: {
        align: 'center',
        layout: 'horizontal',
      },
      title: {
        text: 'Total Payment Break up',
        style: { color: '#0170F3', fontWeight: 900, fontSize: '20px' },
      },
      tooltip: {
        pointFormat: '<b>{point.percentage:.1f}%</b>',
      },
      accessibility: {
        point: {
          valueSuffix: '%',
        },
      },
      plotOptions: {
        pie: {
          allowPointSelect: true,
          cursor: 'pointer',
          showInLegend: true,
        },
      },
      series: [
        {
          name: 'Data',
          colorByPoint: true,
          data: [
            {
              name: 'Principal',
              y: principal,
            },
            {
              name: 'Interest',
              y: interest,
            },
          ],
        },
      ],
    });
  }, [interest, interest, backgroundColor]);

  return (
    <div className="chart" id="container" />
  );
};

export default Pie;
