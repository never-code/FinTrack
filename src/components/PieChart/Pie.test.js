import React from 'react';
import { render } from '@testing-library/react';
import Pie from './Pie';

describe('Pie Graph: ', () => {
  it('should render successfully', () => {
    const { container } = render(<Pie principal={65} interest={35} backgroundColor="#FFFF" />);
    expect(container).toMatchSnapshot();
  });
});
