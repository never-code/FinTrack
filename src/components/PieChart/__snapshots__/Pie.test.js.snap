// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pie Graph:  should render successfully 1`] = `
<div>
  <div
    class="chart"
    data-highcharts-chart="0"
    id="container"
    style="overflow: hidden;"
  >
    <div
      class="highcharts-container "
      dir="ltr"
      id="highcharts-5op6xuy-0"
      style="position: relative; overflow: hidden; width: 600px; height: 400px; text-align: left; line-height: normal; z-index: 0; user-select: none;"
    >
      <svg
        class="highcharts-root"
        height="400"
        style="font-family:\\"Lucida Grande\\", \\"Lucida Sans Unicode\\", Arial, Helvetica, sans-serif;font-size:12px;"
        version="1.1"
        viewBox="0 0 600 400"
        width="600"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc>
          Created with Highcharts 8.1.2
        </desc>
        <defs>
          <clippath
            id="highcharts-5op6xuy-3-"
          >
            <rect
              fill="none"
              height="340"
              width="580"
              x="0"
              y="0"
            />
          </clippath>
        </defs>
        <rect
          class="highcharts-background"
          fill="#FFFF"
          height="400"
          rx="0"
          ry="0"
          width="600"
          x="0"
          y="0"
        />
        <rect
          class="highcharts-plot-background"
          fill="none"
          height="340"
          width="580"
          x="10"
          y="10"
        />
        <rect
          class="highcharts-plot-border"
          data-z-index="1"
          fill="none"
          height="340"
          width="580"
          x="10"
          y="10"
        />
        <g
          class="highcharts-series-group"
          data-z-index="3"
        >
          <g
            class="highcharts-series highcharts-series-0 highcharts-pie-series highcharts-tracker"
            data-z-index="0.1"
            opacity="1"
            style="cursor:pointer;"
            transform="translate(10,10) scale(1 1)"
          >
            <path
              class="highcharts-point highcharts-color-0"
              d="M 290 10 A 160 160 0 0 1 289.8400000266667 10.000079999993318 L 290 170 A 0 0 0 0 0 290 170 Z"
              fill="#FBD94F"
              opacity="1"
              stroke="#ffffff"
              stroke-linejoin="round"
              stroke-width="1"
              transform="translate(0,0)"
            />
            <path
              class="highcharts-point highcharts-color-1"
              d="M 290 10 A 160 160 0 0 1 289.8400000266667 10.000079999993318 L 290 170 A 0 0 0 0 0 290 170 Z"
              fill="#0170F3"
              opacity="1"
              stroke="#ffffff"
              stroke-linejoin="round"
              stroke-width="1"
              transform="translate(0,0)"
            />
          </g>
          <g
            class="highcharts-markers highcharts-series-0 highcharts-pie-series"
            data-z-index="0.1"
            opacity="1"
            transform="translate(10,10) scale(1 1)"
          />
        </g>
        <text
          class="highcharts-title"
          data-z-index="4"
          style="color:#0170F3;font-size:20px;font-weight:900;fill:#0170F3;"
          text-anchor="middle"
          x="300"
          y="25"
        >
          <tspan>
            Total Payment Break up
          </tspan>
        </text>
        <text
          class="highcharts-subtitle"
          data-z-index="4"
          style="color:#666666;fill:#666666;"
          text-anchor="middle"
          x="300"
          y="10"
        />
        <text
          class="highcharts-caption"
          data-z-index="4"
          style="color:#666666;fill:#666666;"
          text-anchor="start"
          x="10"
          y="397"
        />
        <g
          class="highcharts-data-labels highcharts-series-0 highcharts-pie-series highcharts-tracker"
          data-z-index="6"
          opacity="0"
          style="cursor:pointer;"
          transform="translate(10,10) scale(1 1)"
          visibility="hidden"
        >
          <path
            class="highcharts-data-label-connector highcharts-color-0"
            d="M 464.2912395957899 256.2581949505139 C 459.2912395957899 256.2581949505139 443.25312216039924 248.08636595520204 437.90708301526905 245.36242295676476 L 432.56104387013886 242.63847995832748"
            fill="none"
            stroke="#FBD94F"
            stroke-width="1"
          />
          <path
            class="highcharts-data-label-connector highcharts-color-1"
            d="M 115.7087604042101 83.74180504948613 C 120.7087604042101 83.74180504948613 136.7468778396007 91.91363404479797 142.09291698473092 94.63757704323525 L 147.43895612986114 97.36152004167253"
            fill="none"
            stroke="#0170F3"
            stroke-width="1"
          />
          <g
            class="highcharts-label highcharts-data-label highcharts-data-label-color-0"
            data-z-index="1"
            transform="translate(469,246)"
          >
            <text
              data-z-index="1"
              style="color:#000000;cursor:pointer;font-size:11px;font-weight:bold;fill:#000000;"
              x="5"
              y="16"
            >
              <tspan>
                Principal
              </tspan>
            </text>
          </g>
          <g
            class="highcharts-label highcharts-data-label highcharts-data-label-color-1"
            data-z-index="1"
            transform="translate(0,74)"
          >
            <text
              data-z-index="1"
              style="color:#000000;cursor:pointer;font-size:11px;font-weight:bold;fill:#000000;"
              x="5"
              y="16"
            >
              <tspan>
                Interest
              </tspan>
            </text>
          </g>
        </g>
        <g
          class="highcharts-legend"
          data-z-index="7"
          transform="translate(300,362)"
        >
          <rect
            class="highcharts-legend-box"
            fill="none"
            rx="0"
            ry="0"
            visibility="visible"
          />
          <g
            data-z-index="1"
          >
            <g>
              <g
                class="highcharts-legend-item highcharts-pie-series highcharts-color-0"
                data-z-index="1"
                transform="translate(8,3)"
              >
                <text
                  data-z-index="2"
                  style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;"
                  text-anchor="start"
                  x="21"
                  y="15"
                >
                  <tspan>
                    Principal
                  </tspan>
                </text>
                <rect
                  class="highcharts-point"
                  data-z-index="3"
                  fill="#FBD94F"
                  height="12"
                  rx="6"
                  ry="6"
                  width="12"
                  x="2"
                  y="4"
                />
              </g>
              <g
                class="highcharts-legend-item highcharts-pie-series highcharts-color-1"
                data-z-index="1"
                transform="translate(0,3)"
              >
                <text
                  data-z-index="2"
                  style="color:#333333;cursor:pointer;font-size:12px;font-weight:bold;fill:#333333;"
                  text-anchor="start"
                  x="21"
                  y="15"
                >
                  <tspan>
                    Interest
                  </tspan>
                </text>
                <rect
                  class="highcharts-point"
                  data-z-index="3"
                  fill="#0170F3"
                  height="12"
                  rx="6"
                  ry="6"
                  width="12"
                  x="2"
                  y="4"
                />
              </g>
            </g>
          </g>
        </g>
        <text
          class="highcharts-credits"
          data-z-index="8"
          style="cursor:pointer;color:#999999;font-size:9px;fill:#999999;"
          text-anchor="end"
          x="590"
          y="395"
        >
          Highcharts.com
        </text>
      </svg>
    </div>
  </div>
</div>
`;
